### PR TITLE
Clean up some security issues

### PR DESCRIPTION
**Pull request by Mcat12**
**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**

Fix some security issues in the PHP code.

**How does this PR accomplish the above?:**

Remove possible XSS attack vectors and SQL injection sites (change to use parameterized queries).

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: api_db.php [hunk 1]
```diff
@@ -70,7 +70,10 @@ function SQLite3_connect($trytoreconnect)
 	{
 		$from = intval($_GET["from"]);
 		$until = intval($_GET["until"]);
-		$results = $db->query('SELECT timestamp,type,domain,client,status FROM queries WHERE timestamp >= '.$from.' AND timestamp <= '.$until.' ORDER BY timestamp ASC');
+		$stmt = $db->prepare("SELECT timestamp, type, domain, client, status FROM queries WHERE timestamp >= :from AND timestamp <= :until ORDER BY timestamp ASC");
+		$stmt->bindValue(":from", $from);
```
Comment (DL6ER) [inline api_db.php]: I don't really get the point of doing this here. We already have
```
$from = intval($_GET["from"]);
$until = intval($_GET["until"]);
```
so there cannot be any e.g. code injection as everything that is not an integer will be discarded by the above mentioned subroutines. Here is [the definition of `intval()`](http://php.net/manual/en/function.intval.php):
```
int intval ( mixed $var [, int $base = 10 ] )
```
> Returns the integer value of var, using the specified base for the conversion (the default is base 10). intval() should not be used on objects, as doing so will emit an E_NOTICE level error and return 1.

File: api_db.php [hunk 2]
```diff
@@ -87,17 +90,20 @@ function SQLite3_connect($trytoreconnect)
 	$limit = "";
 	if(isset($_GET["from"]) && isset($_GET["until"]))
 	{
-		$limit = "WHERE timestamp >= ".$_GET["from"]." AND timestamp <= ".$_GET["until"];
+		$limit = "WHERE timestamp >= :from AND timestamp <= :until";
```
Comment (DL6ER) [inline api_db.php]: Okay, clearly the `intval()` bit we see above is missing here.

File: api_db.php [hunk 3]
```diff
@@ -274,7 +286,11 @@ function SQLite3_connect($trytoreconnect)
 	}
 
 	// Count permitted queries in intervals
-	$results = $db->query('SELECT (timestamp/'.$interval.')*'.$interval.' interval, COUNT(*) FROM queries WHERE (status != 0 )'.$limit.' GROUP by interval ORDER by interval');
+	$stmt = $db->prepare('SELECT (timestamp/:interval)*:interval interval, COUNT(*) FROM queries WHERE (status != 0 )'.$limit.' GROUP by interval ORDER by interval');
+	$stmt->bindValue(":from", intval($_GET['from']));
+	$stmt->bindValue(":until", intval($_GET['until']));
+	$stmt->bindValue(":interval", $interval);
```
Comment (DL6ER) [inline api_db.php]: `$interval` is safe.

File: api_db.php [hunk 4]
```diff
@@ -253,15 +265,15 @@ function SQLite3_connect($trytoreconnect)
 
 	if(isset($_GET["from"]) && isset($_GET["until"]))
 	{
-		$limit = " AND timestamp >= ".intval($_GET["from"])." AND timestamp <= ".intval($_GET["until"]);
```
Comment (DL6ER) [inline api_db.php]: `intval()` was already used here!

File: api_db.php (hunk 1 above)
Comment (Mcat12) [reply api_db.php]: Mainly for consistency.

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/pi-hole-v3-3-released-its-extra-special/7159/1

**Pull request merged**